File: packages/service-data-generate-image/datastore.py
import boto3
import os
import json

s3 = boto3.client('s3')

bucket_name = os.environ['BUCKET_NAME']
object_key_prefix = 'service-data/'

def put(key, data):
    s3.put_object(Body=json.dumps(data).encode('utf-8'), Bucket=bucket_name, Key=object_key_prefix + key)

def get(key):
    obj = s3.get_object(Bucket=bucket_name, Key=object_key_prefix + key)
    return json.loads(obj['Body'].read())

File: packages/app-image/src/datastore.ts
import * as S3 from 'aws-sdk/clients/s3';

const bucketName = process.env.BUCKET_NAME;
const objectKeyPrefix = 'service-data/'

var s3 = new S3({apiVersion: '2006-03-01'});

export async function get(key: string){
  const data = await s3.getObject({
    Bucket: bucketName!,
    Key: objectKeyPrefix + key
  }).promise();
  if(data === undefined || data.Body === undefined)
    throw Error(`fail to get '${key}' in datastore`);
  return JSON.parse(data.Body!.toString('utf-8'));
}

